docs/DeveloperGuide.md

```diff
@@ -2,14 +2,266 @@
+before calling `printFlashcardList(matchingFlashcards)` to display the found flashcards.
+
+User input is taken to get the index of the flashcard to be removed. `deleteFlashcard` is called from
+class `flashcardList` to remove the flashcard from the original list of flashcards. Finally
+`printSuccessfulDelete` is called from class `Ui` to indicate a successful removal of the flashcard.
+
+The deletion process is now completed and the program will await another command.
+
+An overview of how the Delete operation works is shown with the following sequence diagram
+![DeleteFlashcard.png](umlDiagrams%2FDeleteFlashcard.png)
+
+#### Reason for current implementation
+
+Through using `DeleteCommand` class, which extends `Command` class it increases the level of
+abstraction as the code can now perform the various commands on a class level.
+
+In order to minimise the time for users to search for the flashcard to delete, they are able
+to first search for a sub-list of flashcards with matching questions as the query. This method
+makes the deletion process simple even if the user does not remember the index of the flashcard.
+
+#### Alternative implementation
+
+- Alternative 1: Delete flashcard by index from the start
+    - Pros: Easy to implement and simplifies code
+    - Cons: Cumbersome to delete if user forgets the flashcard's index and has to search
+      through the whole list of flashcards.
+
+### Update Flashcard Feature
+
+#### Current implementation
+
+The current update flashcard feature allows users search for a specific flashcard and update the contents of this
+flashcard.
+It is implemented through the following steps:
+
+Step 1:
+The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2:
+The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Parser`.
+
+In this case, an `UpdateCommand` will be created and returned.
+
+Step 3:
+The `execute()` function of `UpdateCommand` will run, creating an ArrayList of flashcards. Then
+`findFlashcard(flashcards, query)` is called to find flashcards that contain questions or answers that match `query`,
+after which it will call `printFlashCards(matchingFlashcards)` in `UpdateCommand` class to display the flashcards
+found.
+
+The index of the flashcard to be updated is taken from the user input, which is collected by `getUserCommand()`.
+`implementUpdate(flashcards, userText)` is then called to update the question or answer of the flashcard. Finally,
+`printFlashCard(flashcards.get(index))` will be called which prints the flashcard that was updated with its new content.
+
+At this point, the update flashcard process is completed and the program is read to take another command.
+
+An overview of how the Update command works is shown with the following sequence diagram
+![UpdateFlashcard.png](umlDiagrams%2FUpdateFlashcard.png)
```

Comment:
![image](https://user-images.githubusercontent.com/37980219/229274283-3b99bba5-1691-44c7-af55-930c60061c26.png)

Missing line break and extra colon here. Correct example from module website below:

![image](https://user-images.githubusercontent.com/37980219/229274274-5319bb95-6255-440a-9b7c-80c2cc56da35.png)